## Titanic Kaggle Competition / Tutorial


In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.model_selection import cross_val_score
from sklearn.naive_bayes import GaussianNB
from sklearn.linear_model import LogisticRegression
from sklearn.neighbors import KNeighborsClassifier
from pandas import Series,DataFrame

### Read csv file and drop column Name.

In [2]:
df = pd.read_csv('/Users/dg/Titanic/Data/titanic.csv').drop(['Name'], axis=1)
df.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0         0       3    male  22.0                        1   
1         1       1  female  38.0                        1   
2         1       3  female  26.0                        0   
3         1       1  female  35.0                        1   
4         0       3    male  35.0                        0   

   Parents/Children Aboard     Fare  
0                        0   7.2500  
1                        0  71.2833  
2                        0   7.9250  
3                        0  53.1000  
4                        0   8.0500

### Changed columns Sex from cat to factor.

In [3]:
df1 = pd.get_dummies(df)
df1

Survived  Pclass   Age  Siblings/Spouses Aboard  Parents/Children Aboard  \
0           0       3  22.0                        1                        0   
1           1       1  38.0                        1                        0   
2           1       3  26.0                        0                        0   
3           1       1  35.0                        1                        0   
4           0       3  35.0                        0                        0   
..        ...     ...   ...                      ...                      ...   
882         0       2  27.0                        0                        0   
883         1       1  19.0                        0                        0   
884         0       3   7.0                        1                        2   
885         1       1  26.0                        0                        0   
886         0       3  32.0                        0                        0   

        Fare  Sex_female  Sex_male  
0     7.2500           0         1  
1    71.2833           1         0  
2     7.9250           1         0  
3    53.1000           1         0  
4     8.0500           0         1  
..       ...         ...       ...  
882  13.0000           0         1  
883  30.0000           1         0  
884  23.4500           1         0  
885  30.0000           0         1  
886   7.7500           0         1  

[887 rows x 8 columns]

### Splited dataset with sklearn.

In [4]:
X = df1.iloc[:,1:].values
y = df1.iloc[:,0:1].values

In [5]:
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=0.3)

### Created models and compared.

In [6]:
gnb = GaussianNB()
cv = cross_val_score(gnb,X_train,y_train.ravel(),cv=5)
print(cv)
print(cv.mean())

[0.77419355 0.7983871  0.7983871  0.76612903 0.81451613]
0.7903225806451613


In [7]:
lr = LogisticRegression(max_iter = 2000)
cv = cross_val_score(lr,X_train,y_train.ravel(),cv=5)
print(cv)
print(cv.mean())

[0.80645161 0.83064516 0.79032258 0.77419355 0.74193548]
0.7887096774193549


In [8]:
knn = KNeighborsClassifier()
cv = cross_val_score(knn,X_train,y_train.ravel(),cv=5)
print(cv)
print(cv.mean())

[0.66935484 0.72580645 0.75806452 0.7016129  0.73387097]
0.717741935483871


In [9]:
M = {'Model':['Naive Bayes','Logistic Regression','K Nearest Neighbor'],
       'Accuracy Score':[0.7951612903225806,0.8064516129032258,0.7274193548387096]}
df = DataFrame(M)
print(df)

                 Model  Accuracy Score
0          Naive Bayes        0.795161
1  Logistic Regression        0.806452
2   K Nearest Neighbor        0.727419


So the best model is Logistic Regression model whose Accuracy Score is  0.806452.